components/forms/DynamicRow/DynamicRow.test.js

```diff
@@ -216,5 +217,33 @@ describe.each([["en"], ["fr"]])("Generate a dynamic row", (lang) => {
       // values 1,2,3 were deleted with row 1
       expect(fieldValues).toEqual(["3", "4", "5", "6", "7", "8"]);
     });
+
+    test("Maximum number of rows", async () => {
```

Comment:
This test doesn't verify that it prevents the user from adding 4 rows.

Reply:
It does because of `expect(screen.queryAllByTestId("add-row-button-1")).toHaveLength(0);` which verifies that the `Add` doest not exist anymore on the webpage.